File: java/lld/booking/library/BookCopy.java
// A single lendable UNIT — the real inventory.
public class BookCopy {
    public final String barcode, isbn;
    public CopyStatus status = CopyStatus.AVAILABLE;
    public BookCopy(String barcode, String isbn) { this.barcode = barcode; this.isbn = isbn; }
    public boolean isAvailable() { return status == CopyStatus.AVAILABLE; }
}

File: typescript/lld/booking/library/BookCopy.ts
import { CopyStatus } from './enums';

// A single lendable UNIT. This is the real inventory, not the title.
export class BookCopy {
    status: CopyStatus = CopyStatus.AVAILABLE;
    constructor(
        public readonly barcode: string,
        public readonly isbn: string,
    ) {}

    isAvailable(): boolean { return this.status === CopyStatus.AVAILABLE; }
}